### DISCUSSION TITLE
How to manager flux multiple git sources  when primary git source is completely deleted.

### DISCUSSION THREAD

**Discussion by eravindar12** [fluxcd/flux2#4044] [Q&A] [answered]
I need a suggestion on the below scenario - How to manage flux multiple git sources when the primary git source is completely deleted.

**Description:** 
I set it up flux in one of the GKE clusters using the bootstrap Gitlab command for one of the GitLab repo so here, for example, the URL is [gitA.com](http://gita.com/) and the repo name is A and now we're migrating Gitlab to the new separate host so the URL and repo will be different which is [gitlabB.com](http://gitlabb.com/), repo is B
Based on this requirement again I created a new git source as per the new URL [gitB.com](http://gitb.com/) and the repo is B, and migrated the application and created Kustomization path for B tested it all works well with the B git source.
But my question is if I completely decommission the source [gitlabA.com](http://gitlaba.com/), repo A  - am I still able to make it work flux for the repo B or do I need to install/run the bootstrap Gitlab command by using the URL [gitB.com](http://gitb.com/) and repo B?
Because the [gitlabB.com](http://gitlabb.com/) and repo B were added based on the [gitlabA.com](http://gitlaba.com/) and git source A flux-feet.

**Comment by kingdonb** [comment, marked as answer]
If no Flux kustomization still refers to the old `flux-system` gitrepo that was installed with bootstrap, then there will be no immediate consequences for deleting it. However, if you use the bootstrap upgrade procedure, which is the generally recommended process for upgrading, then you can expect Flux to write back over and use the `flux-system` name again. It's better to change the resource place, or if you must create a new resource to better manage a smooth transition or whatever reason, go back and change the original resource, then restore to match the bootstrap config.

You want all of the files that say "DO NOT EDIT" at the top to be pristine so that bootstrapping again will not clobber any of your changes. You can hopefully always bootstrap again, if something goes wrong in this process, so I'm not sure the reason for the second temporary repo. Just make the change in-place, in my opinion, and if something goes wrong, bootstrap again. Hope this helps!

**Comment by eravindar12** [comment]
@kingdonb - Yes and Thanks a lot!
